java/src/main/java/io/keen/client/java/JacksonJsonHandler.java

```diff
@@ -25,7 +29,44 @@
      */
     @Override
     public Map<String, Object> readJson(Reader reader) throws IOException {
-        return mapper.readValue(reader, MAP_TYPE);
+        // TODO : We can't assume the top-level node is a JSON Object anymore, because parts of the
+        // API we access return a JSON Array as the root. So we need to detect the type and decide
+        // what to return, so we need a different return type. Technically it could be a List or
```

Comment:
Yeah I'll replace this TODO with either a clearer NOTE or just create an Issue, but do you have opinions on how to handle the situation in the next major revision? I'd be happy to hear what layer of abstraction you think should handle the branch between top-level object and top-level array.